packages/account/src/providers/transaction-response/transaction-response.ts

```diff
@@ -146,20 +255,14 @@ export class TransactionResponse {
 
       for await (const { statusChange } of subscription) {
         if (statusChange) {
-          this.expectedStatus = statusChange.type;
+          this.status = statusChange;
           break;
         }
       }
 
       return this.fetch();
     }
 
-    // Refetch if the expected status is not the same as the response status
-    if (this.expectedStatus && response.transaction.status?.type !== this.expectedStatus) {
```

Comment:
I removed this polling code added in #2896. The problem was happening because `statusChange` sent a `SuccessStatus` but fetching the whole transaction with receipts would return a transaction in `SubmittedStatus`. Because we're now reading all the malleable fields from the subscription directly and not doing that additional fetch, there's no need for this solution.